index.html

```diff
@@ -812,31 +742,40 @@ <h3>
           Release wake lock algorithm
         </h3>
         <p>
-          To <dfn>release a wake lock</dfn> for a given {{WakeLock}} object
-          |lock:WakeLock|, run these steps <a>in parallel</a>:
+          To <dfn>release a wake lock</dfn> for a given |lockPromise:Promise| and
+          |type:WakeLockType|, run these steps <a>in parallel</a>:
         </p>
         <ol class="algorithm" data-link-for="WakeLock">
-          <li>Let |type:WakeLockType| be |lock|'s <a>type</a>.
+          <li>Reject |lockPromise| with an "{{AbortError}}" {{DOMException}}.
```

Comment:
While not wrong per se, it feels a bit weird that this algorithm rejects any promise passed to it even if it's not in `[[ActiveLocks]]`. Is there any advantage to it over the more familiar "check if it's present -> abort if it's not / reject and remove from `[[ActiveLocks]]` if not" sequence of steps?

Reply:
If this were code I would say that there should be a DCHECK that `lockPromise` is in `[[ActiveLocks]]`. I don't think it should be an actual step in the algorithm because there is no circumstance where this algorithm would be called with a Promise not in the set.

Reply:
While writing some tests I've run into the following sequence of events:

```js
const controller = new AbortController();
const lock = WakeLock.request('screen', { signal: controller.signal });
// User hides the page, page visibility handling causes |lock| to be released
// User shows the page again
controller.abort();
```

Since the spec does not mention anything about removing an algorithm from an `AbortSignal` instance (and Blink does not even support that), having a DCHECK causes a crash rather than just ignoring the request to release the lock.

Reply:
Ignoring Blink-specific limitations for a moment it seems like the desired behavior is that when the page visibility state changes `lock` should be rejected with an error indicating that the wake lock has been released due to the page visibility state change. Calling `controller.abort()` should be a no-op at this point since the lock has been released and the Promise is no longer pending.

Script that wants to reacquire a screen wake lock when the page becomes visible again can do so, and can even use the same AbortController since calling `abort()` will only have an effect on the locks that are still active.

Given @rakuco's example I think my original comment that this algorithm will always be called with a Promise that is in `[[ActiveLocks]]` was incorrect. I'm still not sure it matters to encode that check into the specification because the only time a Promise is removed from `[[ActiveLocks]]` it is when it is rejected and so attempting to reject it again is a no-op. The current patch acknowledges this by aborting the steps later and not requesting the operating system release a lock if the Promise has been resolved.